src/app/core/message-list/message-list.component.ts

```diff
@@ -34,6 +35,79 @@ export class MessageListComponent implements OnInit {
     this.sendMessage();
   }
 
+    quickRoll(r) {
+        let rollStr = ""
+        let rollRE = /^(\/r(oll)? )?(([0-9]+)[dD]([0-9]+)|0)?((kh|kl)1?)?((\+|\-)[0-9]+)? ?(.*)?/;
+        let m = rollRE.exec(this.messageInput);
+        if (this.messageInput != "" && m == null) {
+            return;
+        }
+        rollStr = (m&&m[1])? m[1] : "/roll ";
+
+        let mod=0
+        if (m&&m[9]) {
```

Comment:
There's another way of doing this with optional chaining -
```suggestion
        if (m.?[9]) {
```

Reply:
Thanks, didn’t know about that trick.